### PR TITLE
Feature/error component

### PR DESCRIPTION
**Pull request by jhonatangomes**
To normalize errors and create a pattern to them a new error component is being created. 

Initially only playback errors are being changed to this new format, but errors from other layers can adopt this new format too.

This component will trigger a new error event that contains data that can be useful to, for example, an error screen plugin render information to users.

This new error format contains: 

**code**: the error code
**description**: error description
**level**:  warn or fatal
**origin**: hls, html5, etc
**scope**: playback, container, etc
**raw error**: the error without modification

### CODE REVIEW

File: src/base/events.js [hunk 1]
```diff
@@ -282,7 +282,15 @@ Events.PLAYER_SEEK = 'seek'
  * @event PLAYER_ERROR
  * @param {Object} error the error
  */
-Events.PLAYER_ERROR = 'error'
+Events.PLAYER_ERROR = 'playererror'
+/**
+ * Fired when there is an error
+ *
+ * @event ERROR
+ * @param {Object} error
```
Comment (leandromoreira) [inline src/base/events.js]: Can you describe/document the proposed error object here? Like we do at [Player's constructor](https://github.com/jhonatangomes/clappr/blob/c5e5fcf174c9d60135750608d4e3af5c4f4aa45e/src/components/player.js#L152-L160)?

Comment (jhonatangomes) [reply src/base/events.js]: yes, done.

File: src/playbacks/html5_video/html5_video.js [hunk 2]
```diff
@@ -383,7 +383,15 @@ export default class HTML5Video extends Playback {
   }
 
   _onError() {
-    this.trigger(Events.PLAYBACK_ERROR, this.el.error, this.name)
+    const { code, message } = this.el.error
```
Comment (otmjka) [inline src/playbacks/html5_video/html5_video.js]: there will be an error if `this.el.error` is `null`

Comment (jhonatangomes) [reply src/playbacks/html5_video/html5_video.js]: This PR is going on at https://github.com/clappr/clappr/pull/1614

**Pull request closed without merging**